lucene/suggest/src/java/org/apache/lucene/search/spell/DirectSpellChecker.java

```diff
@@ -317,7 +332,11 @@ public void setDistance(StringDistance distance) {
       SuggestMode suggestMode, float accuracy) throws IOException {
     final CharsRefBuilder spare = new CharsRefBuilder();
     String text = term.text();
-    if (minQueryLength > 0 && text.codePointCount(0, text.length()) < minQueryLength)
+
+    int textLength = text.codePointCount(0, text.length());
+    if (minQueryLength > 0 && textLength < minQueryLength)
+      return new SuggestWord[0];
+    if (maxQueryLength > 0 && textLength > maxQueryLength)
```

Comment:
Here we could group the "if" check, remove the maxQueryLength > 0 condition (with maxQueryLength initialized to Integer.MAX_VALUE), and have only one return empty array because this is the same semantic (and add brackets).

Reply:
Yes to both!